ctsit/nacculator
# Add documentation, delete redundant files, repair unit test, and add more specificity to error log

This change introduces several small maintenance updates to nacculator: 
I have added documentation for `get_subject_list.js` to the README and modified some files to make the formatting more consistent (such as in the csf form's `builder.py` file and the uds3 folder's `__init__.py` file). 
I deleted redundant files that were no longer being used anywhere in the code or workflow (`getPacketStatus.js` and `split_ivp_fvp.sh`).
I added more specificity to error logging, so that the form location of an invalid field would be listed either in the error message or its callback. This will aid nacculator users who work in REDCap because REDCap does not list fieldnames during data entry. It will help save users from having to dig through the official data element dictionary for reference.
I repaired the old `test_m1.py` unit test so that we can keep better track of the set_zeros_to_blanks function within `redcap2nacc` and the milestone form builder.

This change was made to address [issue 73](https://github.com/ctsit/nacculator/issues/73) along with some quality of life upgrades that I made note of as I have used the program.


_List the steps needed to make sure this thing works. Be precise._

1. From the command-line (assuming you have installed nacculator, acquired a sample data .csv, and saved a config file), run:
`python3 -m unittest`
2. The output should look like this: 59 passing unit tests (including the four repaired milestone unit tests).
3. Next, run:
`redcap2nacc -ivp <data.csv >data.txt`
4. Observe that the program works as normal, and any returned errors now include the form that contains the offending field either in the log message or the callback message.

- Verify the thing does this: converts nacc forms and data from .csv to .txt format
- Verify the thing does not do that: print incorrect data, nor print data incorrectly


## Affirmations

All of these should have a check by them. Any exception requires an explanation.
